app/models/release_platform_run.rb

```diff
@@ -201,7 +201,8 @@ def last_commit
   end
 
   def finished_steps?
-    last_commit&.step_runs&.where(release_platform_run: self)&.success&.size == steps.size
+    return false if release_platform.release_step.blank?
```

Comment:
had to do this because a lot of tests, especially under step level, create steps outside of a train and may not have a `release_step` which can blow up the below query.